core/src/test/scala/unit/kafka/server/epoch/util/ReplicaFetcherMockBlockingSend.scala

```diff
@@ -40,9 +42,15 @@ import scala.collection.Map
   * setOffsetsForNextResponse
   */
 class ReplicaFetcherMockBlockingSend(offsets: java.util.Map[TopicPartition, EpochEndOffset],
-                                     sourceBroker: BrokerEndPoint,
+                                     override private[server] val sourceBroker: BrokerEndPoint,
                                      time: Time)
-  extends BlockingSend {
+  extends BrokerBlockingSender(sourceBroker = sourceBroker,
```

Comment:
This seems to defeat the purpose of having the BlockingSend trait. Perhaps, it's better to add brokerEndPoint() to the BlockingSend trait and still have this class implement BlockingSend?

Reply:
Would there ever be a scenario in which we wouldn't want `brokerEndPoint` to be a part of the `BlockingSend` trait? I.e., is there ever an implementation of `BlockingSend` in which a `brokerEndPoint` would be extraneous?

Reply:
The AbstractFetcherThread is tied to an endPoint. If there is no endPoint, the logic probably won't go through AbstractFetcherThread.